freedomofpress/securedrop
# Improve testing of static assets in Debian packaging

## Status

Ready for review

## Description of Changes

Add more complete testing of packaging of static assets.

Simplify conftest, eliminate testutils in builder-xenial scenario.

In builder-xenial/test_securedrop_deb_package, eliminate some host calls and inefficient parameterization, to speed things up.

Fixes #5643.

## Testing

- Check out `develop`.
- Remove the following lines from `securedrop/.rsync-filter`:
```
include static/icons/
include static/icons/**
```
- `make build-debs` -- the missing icons should cause no errors.
- `git checkout -b improve-asset-tests origin/improve-asset-tests`
- `make build-debs` -- the missing icons should cause test errors.

## Checklist

### If you made changes to the server application code:


### If you made non-trivial code changes:


Choose one of the following:
